lib/testers/rule-tester.js

```diff
@@ -138,7 +138,17 @@ const IT = Symbol("it");
  * @returns {any} Returned value of `method`.
  */
 function defaultHandler(text, method) {
-    return method.apply(this);
+    try {
+        return method.apply(this);
+    } catch (e) {
+
+        // options.message should not be passed, otherwise the diff message will be lost.
+        throw new assert.AssertionError({
+            actual: e.actual,
+            expected: e.expected,
+            operator: e.operator
+        });
```

Comment:
I had been envisioning something more like this:

```js
try {
    return method.apply(this);
} catch (err) {
    if (err instanceof assert.AssertionError) {
        throw Object.assign({}, new assert.AssertionError, { message: `${e.actual} ${e.operator} ${e.expected}` })
    }
    throw err;
}
```

That way, any unrelated errors (e.g. crashes in a rule) would still get passed through unmodified, and the original message (e.g. "output is incorrect") would still be present for additional context.

Reply:
@not-an-aardvark using a plain object as error object is not the same as throwing a new `AssertionError`. It would not contain the `stack` or `message` as those properties are non-enumerable.